app-directory/share-state/app/page.tsx

```diff
@@ -0,0 +1,178 @@
+          and exports a <Code>useCounter</Code> hook to interact with that
+          value.
+        </Text>
+        <Snippet>
+          {`import { createContext, useContext, useState } from "react";
+
+const CounterContext = createContext({});
+
+function CounterProvider({ children }) {
+  const state = useState(0);
+
+  return <CounterContext.Provider value={state}>{children}</CounterContext.Provider>;
+}
+
+function useCounter() {
+  return useContext(CounterContext);
+}
+
+export { CounterProvider, useCounter };`}
+        </Snippet>
+        <Text>
+          Now, lets wrap our root layout with this context provider. This will
+          make the <Code>useCounter</Code> hook available to all the pages
+          inside this layout.
+        </Text>
+        <Snippet>
+          {`"use client";
+
+import { Link } from "next/link";
+import { CounterProvider, useCounter } from "./context";
+
+function RootLayout({ children }) {
+  const [counter] = useCounter();
+
+  return (
+    <>
+      <nav>
+        <Link href="/">Index</Link>
+        <Link href="/nested">Nested</Link>
+      </nav>
+      {children}
+      <hr />
+      <h2>Counter</h2>
+      <p>{counter}</p>
+    </>
+  );
+}
+
+export default function RootLayoutContainer(props) {
+  return (
+    <CounterProvider>
+      <RootLayout {...props} />
+    </CounterProvider>
+  );
+}`}
+        </Snippet>
+        <Text>
+          Now, lets create a counter component that will allow us to increment
+          and decrement the counter value.
+        </Text>
+        <Snippet>
+          {`"use client";
+
+import { useCounter } from "./context";
+
+export default function Counter() {
+  const [counter, setCounter] = useCounter();
+
+  return (
+    <div>
+      <button onClick={() => setCounter((counter) => counter - 1)}>Decrement</button>
+      <p>{counter}</p>
+      <button onClick={() => setCounter((counter) => counter + 1)}>Increment</button>
+    </div>
+  );
+}`}
+        </Snippet>
+        <Text>
+          And finally we will create an index and a nested page with the same
+          content that will render the counter component.
+        </Text>
+        <Snippet>
+          {`import Counter from "./counter"; 
+
+export default function IndexPage() {
+  return (
+    <div>
+      <h1>Index page</h1>
+      <Counter />
+    </div>
+  );
+}`}
+        </Snippet>
+        <Snippet>
+          {`import Counter from "../counter"; 
+
+export default function NestedPage() {
+  return (
+    <div>
+      <h1>Nested page</h1>
+      <Counter />
+    </div>
+  );
+}`}
+        </Snippet>
+      </section>
+
+      <section className="flex flex-col gap-3">
+        <Text variant="h2">Fetching data from the layout</Text>
+        <Text>
+          If you need your layout to fetch information from the server, you can
+          create a group and make the layout inside the group a client component
+          instead of the root layout
```

Comment:
This says "and make the layout inside the group a client component", but if the group is `(app)` then it doesn't have a layout.

Reply:
I forgot to add it to the file table, thanks!